compiler/optimizer/IDT.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "infra/Cfg.hpp"
+#include "infra/deque.hpp"
+#include "env/TypedAllocator.hpp"
+#include "env/Region.hpp"
+#include "Inliner.hpp"
+
+class IDTNode;
+
+class IDT {
+  friend class IDTNode;
+private:
+  TR::Region *_mem;
+  TR_InlinerBase *_inliner;
+  int _max_idx = -1;
+  IDTNode *_root;
+  int
+  next_idx();
+  // Returns NULL if 0 or > 1 children
+  static IDTNode *
+  getOnlyChild(IDTNode *owner);
```

Comment:
Why aren't these functions on IDTNode?

Reply:
I needed to call them from IDT node for one of the methods. This was my workaround for not being able to call private methods from a friend class